src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -410,6 +410,30 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         auto pfnScrollPositionChanged = std::bind(&TermControl::_TerminalScrollPositionChanged, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3);
         _terminal->SetScrollPositionChangedCallback(pfnScrollPositionChanged);
 
+        // Set up blinking cursor
+        DispatcherTimer cursorTimer;
+        cursorTimer.Interval(std::chrono::milliseconds(500));
+        auto registrationtoken = cursorTimer.Tick({ this, &TermControl::_BlinkCursor });
+
+        _controlRoot.GotFocus([cursorTimer](auto&, auto&) {
+            // Start blinking the cursor when the window is focused.
+            cursorTimer.Start();
+        });
+
+        _controlRoot.LostFocus([this, cursorTimer](auto&, auto&) {
+            // Stop blinking the cursor when the window goes out of focus,
+            // and hide it.
+            cursorTimer.Stop();
+            _terminal->SetCursorVisible(false);
+        });
+
+        _controlRoot.KeyDown([this, cursorTimer](auto&, auto&) {

```

Comment:
Could we do this in the `TermControl::_KeyDownHandler` we already have registered?